### DISCUSSION TITLE
Navigation Midpoints Not Saved To Tracks

### DISCUSSION THREAD

**Discussion by Sonwon1** [osmandapp/OsmAnd#17974] [Q&A] [answered]
Is this normal or planned behavior to not save the names of the navigation midpoints to the GPX files? It does show the waypoints on the map however the name is lost.

**Comment by DmitryAlexei** [comment]
@Sonwon1 Please add steps to reproduce your issue
how do you add waypoints? how do you save your track?

**Reply by Sonwon1** [reply, marked as answer]
I am going to close this. Here is the answer, yes when you save the navigation route into a GPX file when you reload the file none of the mid points will show in the navigation list. However if you look at the route on the map you will see the points and if you touch them they will open showing the name. I already opened a feature request to have them show and to just save the navigation list and restore from the list.

Also, the best way to create a multi stop navigation route is to create map markers for each point and then build the navigation list from the Markers.
